File: sdk/python/aragora_sdk/namespaces/status.py
"""
Status Namespace API

Provides methods for system status and health:
- Service health checks
- System metrics
- Operational status
"""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from ..client import AragoraAsyncClient, AragoraClient


class StatusAPI:
    """Synchronous Status API."""

    def __init__(self, client: AragoraClient):
        self._client = client

    def get_status(self) -> dict[str, Any]:
        """Get overall system status."""
        return self._client.request("GET", "/api/v1/status")

    def get_components(self) -> dict[str, Any]:
        """Get status of individual components."""
        return self._client.request("GET", "/api/v1/status/components")

    def get_history(self) -> dict[str, Any]:
        """Get status history."""
        return self._client.request("GET", "/api/v1/status/history")

    def get_incidents(self) -> dict[str, Any]:
        """Get current and recent incidents."""
        return self._client.request("GET", "/api/v1/status/incidents")

    def get_summary(self) -> dict[str, Any]:
        """Get status summary."""
        return self._client.request("GET", "/api/v1/status/summary")

    def get_uptime(self) -> dict[str, Any]:
        """Get uptime percentages and current uptime details."""
        return self._client.request("GET", "/api/v1/status/uptime")

    def get_public_surfaces(self) -> dict[str, Any]:
        """Get public surface readiness inventory."""
        return self._client.request("GET", "/api/v1/public/surfaces")


class AsyncStatusAPI:
    """Asynchronous Status API."""

    def __init__(self, client: AragoraAsyncClient):
        self._client = client

    async def get_status(self) -> dict[str, Any]:
        """Get overall system status."""
        return await self._client.request("GET", "/api/v1/status")

    async def get_components(self) -> dict[str, Any]:
        """Get status of individual components."""
        return await self._client.request("GET", "/api/v1/status/components")

    async def get_history(self) -> dict[str, Any]:
        """Get status history."""
        return await self._client.request("GET", "/api/v1/status/history")

    async def get_incidents(self) -> dict[str, Any]:
        """Get current and recent incidents."""
        return await self._client.request("GET", "/api/v1/status/incidents")

    async def get_summary(self) -> dict[str, Any]:
        """Get status summary."""
        return await self._client.request("GET", "/api/v1/status/summary")

    async def get_uptime(self) -> dict[str, Any]:
        """Get uptime percentages and current uptime details."""
        return await self._client.request("GET", "/api/v1/status/uptime")

    async def get_public_surfaces(self) -> dict[str, Any]:
        """Get public surface readiness inventory."""
        return await self._client.request("GET", "/api/v1/public/surfaces")

File: sdk/typescript/src/namespaces/status.ts
/**
 * Status Namespace API
 *
 * Provides endpoints for the platform status page including
 * service health, incident history, and maintenance windows.
 */

import type { AragoraClient } from '../client';

/** Overall platform status */
export type PlatformStatus = 'operational' | 'degraded' | 'partial_outage' | 'major_outage';

/** Service component status */
export interface ServiceComponent {
  id: string;
  name: string;
  status: PlatformStatus;
  description?: string;
  updated_at: string;
}

/** Status page incident */
export interface StatusIncident {
  id: string;
  title: string;
  status: 'investigating' | 'identified' | 'monitoring' | 'resolved';
  impact: PlatformStatus;
  message: string;
  components: string[];
  created_at: string;
  resolved_at?: string;
}

/** Maintenance window */
export interface MaintenanceWindow {
  id: string;
  title: string;
  description: string;
  scheduled_start: string;
  scheduled_end: string;
  components: string[];
  status: 'scheduled' | 'in_progress' | 'completed';
}

/** Status page summary */
export interface StatusSummary {
  status: PlatformStatus;
  components: ServiceComponent[];
  active_incidents: StatusIncident[];
  upcoming_maintenance: MaintenanceWindow[];
  updated_at: string;
}

/** Public surface readiness state */
export interface PublicSurfaceReadiness {
  id: string;
  name: string;
  readiness: 'live' | 'partial';
  paths: string[];
  message: string;
  backend_conditional: boolean;
  placeholder_backed: boolean;
  details: Record<string, unknown>;
}

/** Public surface readiness inventory response */
export interface PublicSurfacesResponse {
  data: {
    surfaces: PublicSurfaceReadiness[];
    summary: {
      total: number;
      live: number;
      partial: number;
    };
  };
}

/**
 * Status namespace for platform health monitoring.
 *
 * @example
 * ```typescript
 * const summary = await client.status.getSummary();
 * console.log(`Platform is ${summary.status}`);
 * ```
 */
export class StatusNamespace {
  constructor(private client: AragoraClient) {}

  /** Get overall status summary. */
  async getSummary(): Promise<StatusSummary> {
    return this.client.request<StatusSummary>('GET', '/api/v1/status');
  }

  /** List service components and their status. */
  async listComponents(): Promise<ServiceComponent[]> {
    const response = await this.client.request<{ components: ServiceComponent[] }>(
      'GET',
      '/api/v1/status/components'
    );
    return response.components;
  }

  /** List status incidents. */
  async listIncidents(options?: {
    limit?: number;
    status?: string;
  }): Promise<StatusIncident[]> {
    const response = await this.client.request<{ incidents: StatusIncident[] }>(
      'GET',
      '/api/v1/status/incidents',
      { params: options }
    );
    return response.incidents;
  }

  /** List upcoming and active maintenance windows. */
  async listMaintenance(): Promise<MaintenanceWindow[]> {
    const response = await this.client.request<{ maintenance: MaintenanceWindow[] }>(
      'GET',
      '/api/v1/status/incidents'
    );
    return response.maintenance;
  }

  /** Get uptime summary and historical uptime periods. */
  async getUptime(): Promise<Record<string, unknown>> {
    return this.client.request<Record<string, unknown>>('GET', '/api/v1/status/uptime');
  }

  /** Get public surface readiness inventory. */
  async getPublicSurfaces(): Promise<PublicSurfacesResponse> {
    return this.client.request<PublicSurfacesResponse>('GET', '/api/v1/public/surfaces');
  }
}
